ogre2/src/Ogre2RenderEngine.cc

```diff
@@ -89,8 +89,8 @@ Ogre2RenderEngine::Ogre2RenderEngine() :
 
 #ifdef __APPLE__
   // on OSX the plugins may be placed in the parent lib directory
-  if (ogrePath.rfind("OGRE") == ogrePath.size()-4u)
-    this->ogrePaths.push_back(ogrePath.substr(0, ogrePath.size()-5));
+//   if (ogrePath.rfind("OGRE") == ogrePath.size()-4u)
+//     this->ogrePaths.push_back(ogrePath.substr(0, ogrePath.size()-5));
```

Comment:
?

Reply:
This conditional block in `Ogre2RenderEngine.cc` causes a segfault because the brew installation of Ogre2 only places the plugin libraries in `/usr/local/opt/ogre2.1/lib/OGRE-2.1/OGRE` and not `/usr/local/opt/ogre2.1/lib/OGRE-2.1`. If the second location is included as well then Ogre will throw an exception when it fails to find anything in that directory. This exception is not caught in `ignition-rendering` and causes the segfault.

It might be better to remove the entire block, but it's probably better to split the PR into just the example (new code) and other changes that are macOS specific.